habitat-lab/habitat/core/utils.py

```diff
@@ -114,25 +115,30 @@ def center_crop(obs, new_shape):
     return obs
 
 
-class DatasetFloatJSONEncoder(json.JSONEncoder):
-    r"""JSON Encoder that sets a float precision for a space saving purpose and
-    encodes ndarray and quaternion. The encoder is compatible with JSON
-    version 2.0.9.
-    """
-
+class DatasetJSONEncoder(json.JSONEncoder):
     def default(self, obj):
-        # JSON doesn't support numpy ndarray and quaternion
         if isinstance(obj, np.ndarray):
             return obj.tolist()
         if isinstance(obj, quaternion.quaternion):
             return quaternion_to_list(obj)
+        if isinstance(obj, DictConfig):
+            return dict(obj)
+        if isinstance(obj, ListConfig):
+            return list(obj)
```

Comment:
```suggestion
        if OmegaConf.is_config(obj)
            return OmegaConfg.to_container(obj)
```
This should be much more efficient.